torchtitan/utils.py

```diff
@@ -36,6 +36,26 @@ def _warn_overwrite_env(env, val):
     os.environ[env] = val
 
 
+def set_determinism(seed: Optional[int]) -> None:
+    """
+    Set Python, PyTorch, CUDA seeds and cudnn settings for reproducibility
+    """
+    if seed is not None:
+        # CPU determinism
+        torch.manual_seed(seed)
```

Comment:
Seems this sets the seed for all devices, not just CPU: https://pytorch.org/docs/stable/generated/torch.manual_seed.html

Reply:
Good to know! Yes let's remove redundant `torch.cuda.manual_seed_all(seed)` here.

Reply:
removed